### ISSUE TITLE
Questions on the self-normalizing property

### ISSUE THREAD

**Issue by Karbo123** [bioinf-jku/SNNs#30]
I think the proposed SELU is an powerful non-linearity for MLPs. The self-normalizing property comes from the derivation of the forward propagation. This property could be confirmed by the following codes.

``` python
import torch
f = torch.nn.functional.selu
x = torch.randn(1024, 1024) * 456 + 123
lin = torch.nn.Linear(1024, 1024, bias=False)
_ = torch.nn.init.kaiming_normal_(lin.weight, nonlinearity="linear")
with torch.no_grad():
    for i in range(100): x = f(lin(x))
print(f"mean = {x.mean()}") # 0.00253
print(f"var = {(x ** 2).mean()}") # 1.05135
```

However, **the self-normalizing property only holds for the forward pass, it does not hold for the backward pass** (is it right?). Noisy gradients will definitely be harmful for the learning processes. The proposed SELU is based on ELU, which is based on a selective preference. I wonder if there could exists a more general non-linearity that has self-normalizing properties for both the forward and backward propagations. If it is possible, how could we find it?

**Comment by gklambauer** [comment]
It is also possible to have self-normalization in the backward pass, but you have to trade-off the zero-mean (bias shift) property in the forward pass. See "Bidirectionally self-normalizing neural networks" here: https://arxiv.org/abs/2006.12169
